packages/@sanity/migrate/src/runner/dryRun.ts

```diff
@@ -49,31 +39,12 @@ export async function dryRun(config: MigrationRunnerOptions, migration: Migratio
 
   const context = createBufferFileContext(createReader)
 
-  const mutations = tap(
-    collectMigrationMutations(
-      migration,
-      () => parse(decodeText(streamToAsyncIterator(createReader())), {parse: safeJsonParser}),
-      context,
-    ),
-    (muts) => {
-      stats.currentTransactions = arrify(muts)
-      config.onProgress?.({
-        ...stats,
-        mutations: ++stats.mutations,
-      })
-    },
+  yield* collectMigrationMutations(
```

Comment:
Ooh, nice - I'd completely forgotten about `yield*`. Think there's potential for some simplifications here and there by using it